src/timeseriesflattener/flattened_dataset.py

```diff
@@ -454,49 +454,68 @@ def _get_temporal_feature(
         concatenation.
         """
-        for _ in range(5000):
-            random_index = random.randint(0, len(dfs[0]) - 1)
-            for feature_df in dfs[1:]:
-                if dfs[0].index[random_index] != feature_df.index[random_index]:
-                    raise ValueError(
-                        "Dataframes are not of identical index. Were they correctly aligned before concatenation?",
-                    )
+        base_df = dfs[0]
+        base_length = len(dfs[0])
+        n_dfs = len(dfs)
+
+        log.info(
+            "Checking that dataframes are ready for concatenation - namely that their indeces are aligned. This is a sanity check, and should not be necessary if the dataframes were correctly aligned before concatenation. However, any errors here will completely break predictions, so rather safe than sorry. Can take a while for a large number of dataframes, e.g. 2 minutes for 1_000 dataframes with 2_000_000 rows.",
+        )
+
+        for i, feature_df in enumerate(dfs[1:]):
+            log.debug(f"Checking df {i+2} of {n_dfs}")
+
+            errors = []
+
+            # Check that dataframes are of equal length
+            log.debug("Checking that dataframes are of equal length")
+            if not len(feature_df) == base_length:
+                errors.append(
+                    "Dataframes are not of equal length. ",
+                )
+
+            # Check 5 evenly spaced slices of the dataframe
+            log.debug("Checking that indeces are aligned")
+            if not all(
+                feature_df.index == base_df.index,
+            ):
+                errors.append(
+                    "Dataframes are not aligned. ",
+                )
+
+            if errors:
+                debug_info = f"Columns in dataframes: 0_df: {dfs[0].columns}, feature_df: {feature_df.columns}. Were they correctly aligned before concatenation?"
+                raise ValueError(
+                    f"Dataframes are not ready for concatenation. {errors}, {debug_info}",
+                )
 
     def _concatenate_flattened_timeseries(
         self,
         flattened_predictor_dfs: list[pd.DataFrame],
     ) -> None:
         """Concatenate flattened predictor dfs."""
-        log.info(
-            "Starting concatenation. Will take some time on performant systems, e.g. 30s for 100 features. This is normal.",
-        )
 
         start_time = time.time()
 
         # Check that dfs are ready for concatenation. Concatenation doesn't merge on IDs, but is **much** faster.
         # We thus require that a) the dfs are sorted so each row matches the same ID and b) that each df has a row
         # for each id.
-        self._check_dfs_have_identical_indexes(dfs=flattened_predictor_dfs)
-        self._check_dfs_have_same_lengths(dfs=flattened_predictor_dfs)
+        log.info(
+            "Checking that dfs are ready for concatenation. This will take some time.",
```

Comment:
Redundant, as `_check_dfs_are_ready_for_concat` prints more or less the same. I suggest removing the message here and keeping it in the other method